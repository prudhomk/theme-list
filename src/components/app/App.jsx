import React from 'react';
import CharacterList from '../characters/CharacterList';
import ThemeToggle from '../themes/ThemeToggle';


export default function App() {
  return (
    <>
      <ThemeToggle />
      <CharacterList />
    </>
  );
}
