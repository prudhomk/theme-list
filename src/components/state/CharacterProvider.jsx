import React, { createContext, useContext, useEffect, useState} from 'react';
import { fetchFuturamaCharacters } from '../services/futuramaApi';

const CharacterContext = createContext();

export const CharacterProvider = ({ children }) => {
  const [characters, setCharacters] = useState([]);
  const [selectedApi, setSelectedApi] = useState('futurama');

  const apiMap = {
    futurama: fetchFuturamaCharacters,
  };
}