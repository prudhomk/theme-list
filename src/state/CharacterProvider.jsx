import React, { createContext, useContext, useEffect, useState } from 'react';
import { fetchFuturamaCharacters } from '../services/futuramaApi';

const CharacterContext = createContext();

export const CharacterProvider = ({ children }) => {
  const [characters, setCharacters] = useState([]);
  const [selectedApi, setSelectedApi] = useState('futurama');

  const apiMap = {
    futurama: fetchFuturamaCharacters,
  };

  useEffect(() => {
    apiMap[selectedApi]().then(setCharacters);
  }, [selectedApi]);

  return (
    <CharacterContext.Provider value={{ characters, setSelectedApi, apiMap }}>
      {children}
    </CharacterContext.Provider>
  );
};

export const useCharacters = () => {
  const { characters } = useContext(CharacterContext);
  return characters;
};

export const useSetSelectedApi = () => {
  const { setSelectedApi } = useContext(CharacterContext);
  return setSelectedApi;
};

export const useAvailableAPIs = () => {
  const { apiMap } = useContext(CharacterContext);
  return Object.keys(apiMap);
};
