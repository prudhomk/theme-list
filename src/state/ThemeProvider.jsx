import React, { createContext, useContext, useEffect, useState } from 'react';

const ThemeContext = createContext();

export const ThemeProvider = ({ children }) => {
  const [themes, setThemes] = useState();
  const [selectedTheme, setSelectedTheme] = useState('light');

  return (
    <ThemeContext.Provider value={{ themes, setSelectedTheme }}>
      {children}
    </ThemeContext.Provider>
  );
};



export const useThemes = () => {
  const { themes } = useContext(ThemeContext);
  return themes;
};

export const useSetSelectedTheme = () => {
  const { setSelectedTheme} = useContext(ThemeContext);
  return setSelectedTheme;
};