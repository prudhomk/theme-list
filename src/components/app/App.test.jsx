/* eslint-disable max-len */
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { CharacterProvider } from '../../state/CharacterProvider';
import App from './App';

describe('App component', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    const { asFragment } = render(<CharacterProvider><App /></CharacterProvider>);
    expect(asFragment()).toMatchSnapshot();
  });
});
