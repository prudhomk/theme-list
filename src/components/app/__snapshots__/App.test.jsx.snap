// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <div
    class="Header"
  >
    <h1>
      Futurama List, now with Themes!
    </h1>
    <button>
      Change Theme
    </button>
  </div>
  <ul
    class="Character"
  />
</DocumentFragment>
`;
